src/test/java/bio/terra/service/snapshot/SnapshotServiceTest.java

```diff
@@ -0,0 +1,179 @@
+package bio.terra.service.snapshot;
+
+import bio.terra.common.category.Unit;
+import bio.terra.model.SnapshotAccessInfoModel;
+import bio.terra.model.SnapshotAccessInfoModelBigQuery;
+import bio.terra.model.SnapshotAccessInfoModelBigQueryTable;
+import bio.terra.model.SnapshotModel;
+import bio.terra.model.SnapshotRequestAccessInclude;
+import bio.terra.model.TableModel;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.filedata.google.firestore.FireStoreDependencyDao;
+import bio.terra.service.job.JobService;
+import bio.terra.service.resourcemanagement.google.GoogleProjectResource;
+import bio.terra.service.tabulardata.google.BigQueryPdao;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.time.Instant;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.when;
+
+@Category(Unit.class)
+public class SnapshotServiceTest {
+
+    private static final String SNAPSHOT_NAME = "snapshotName";
+    private static final String SNAPSHOT_DESCRIPTION = "snapshotDescription";
+    private static final String SNAPSHOT_DATA_PROJECT = "tdrdataproject";
+    private static final String SNAPSHOT_TABLE_NAME = "tableA";
+
+
+    @Mock
+    private JobService jobService;
+    @Mock
+    private DatasetService datasetService;
+    @Mock
+    private FireStoreDependencyDao dependencyDao;
+    @Mock
+    private BigQueryPdao bigQueryPdao;
+    @Mock
+    private SnapshotDao snapshotDao;
+
+    private UUID snapshotId;
+    private UUID snapshotTableId;
+    private UUID profileId;
+    private Instant createdDate;
+
+    private SnapshotService service;
+
+    @Before
+    public void setUp() throws Exception {
+        MockitoAnnotations.initMocks(this);
+        service = new SnapshotService(jobService, datasetService, dependencyDao, bigQueryPdao, snapshotDao);
```

Comment:
Is there a reason to prefer using mocks this way rather than using `@MockBean` and injecting `SnapshotService` in this class?

Reply:
Honestly, from what I've seen, using Mockbean requires bringing up the spring wiring for the unit tests which adds a ton of overhead.  As we do more of these tests, I think that will become costly

Reply:
Ok, I didn't know the spring overhead was great, in tests I've written it's been pretty quick. But maybe the overhead is proportional to the number of spring annotations in the project, etc.